workflows/funcscan.nf

```diff
@@ -80,42 +66,69 @@ workflow FUNCSCAN {
     ch_input = Channel.fromSamplesheet("input")
 
     // Some tools require uncompressed input
-    fasta_prep = ch_input
-        .branch {
-            compressed: it[1].toString().endsWith('.gz')
-            uncompressed: it[1]
-        }
+    ch_input_prep = ch_input
+                        .map{ meta, fasta, faa, gbk -> [meta, [fasta, faa, gbk]] }
+                        .transpose()
+                        .branch {
+                            compressed: it[1].toString().endsWith('.gz')
+                            uncompressed: it[1]
+                        }
 
-    GUNZIP_FASTA_PREP ( fasta_prep.compressed )
-    ch_versions = ch_versions.mix( GUNZIP_FASTA_PREP.out.versions )
+    GUNZIP_INPUT_PREP ( ch_input_prep.compressed )
+    ch_versions = ch_versions.mix( GUNZIP_INPUT_PREP.out.versions )
 
     // Merge all the already uncompressed and newly compressed FASTAs here into
     // a single input channel for downstream
-    ch_unzipped_fastas = GUNZIP_FASTA_PREP.out.gunzip
-                        .mix( fasta_prep.uncompressed )
-
-    // Split each FASTA into long and short contigs to
-    // speed up BGC workflow with BGC-compatible contig lengths only
-    SEQKIT_SEQ_LONG ( ch_unzipped_fastas )
-    SEQKIT_SEQ_SHORT ( ch_unzipped_fastas )
-    ch_versions = ch_versions.mix( SEQKIT_SEQ_LONG.out.versions )
-    ch_versions = ch_versions.mix( SEQKIT_SEQ_SHORT.out.versions )
-
-    ch_prepped_input_long = SEQKIT_SEQ_LONG.out.fastx
-                                .map{ meta, file -> [ meta + [id: meta.id + '_long', length: "long" ], file ] }
-                                .filter{
-                                    meta, fasta ->
-                                        !fasta.isEmpty()
+    ch_intermediate_input = GUNZIP_INPUT_PREP.out.gunzip
+                                .mix( ch_input_prep.uncompressed )
+                                .groupTuple()
+                                .map{
+                                    meta, files ->
+                                        def fasta_found   = files.find{it.toString().tokenize('.').last().matches('fasta|fas|fna|fa')}
+                                        def faa_found     = files.find{it.toString().endsWith('.faa')}
+                                        def gbk_found     = files.find{it.toString().tokenize('.').last().matches('gbk|gbff')}
+                                        def fasta         = fasta_found   != null ? fasta_found   : []
+                                        def faa           = faa_found     != null ? faa_found     : []
+                                        def gbk           = gbk_found     != null ? gbk_found     : []
```

Comment:
So this  PR also includes the addition of preannotated input right?

Reply:
Yes, will have to do both in one because I'm a dumdum and commited them and don't want to roll these back too..